ecommerce/frontend/src/components/checkout/EditCardModal.js

```diff
@@ -0,0 +1,157 @@
+import { useState } from 'react'
+import Cards from 'react-credit-cards'
+import 'react-credit-cards/es/styles-compiled.css'
+import { Modal, Form, Input } from 'antd'
+
+export const EditCardModal = ({
+    cardProps = {
+        cvc: '',
+        expiry: '',
+        number: '',
+        name: '',
+    },
+    cardId = {},
+    cardName = '',
+    visible = false,
+    setVisible = () => {},
+}) => {
+    /* This might not remounted after closing and opening again, be careful */
+    const [cvc, setCvc] = useState(cardProps.cvc)
+    const [expiry, setExpiry] = useState(cardProps.expiry)
+    const [name, setName] = useState(cardProps.name)
+    const [number, setNumber] = useState(cardProps.number)
+    const [cardNickname, setCardNickname] = useState(cardName)
+    const [focused, setFocused] = useState('')
+
+    const [form] = Form.useForm()
+    const formItemLayout = {
+        labelCol: {
+            xs: {
+                span: 24,
+            },
+            sm: {
+                span: 6,
+            },
+        },
+        wrapperCol: {
+            xs: {
+                span: 24,
+            },
+            sm: {
+                span: 18,
+            },
+        },
+    }
+
+    const handleInputFocus = val => setFocused(val.target.name)
+
+    return (
+        <Modal
+            visible={visible}
+            title="Edit your credit card information"
+            destroyOnClose
+            onOk={() => {
+                setVisible(false)
+            }}
+            onCancel={() => {
+                setVisible(false)
+            }}
+            cancelText="Go Back"
+            okText="Edit">
+            <Form
+                {...formItemLayout}
+                layout="horizontal"
+                form={form}
+                name="editCardForm"
+                onFinish={() => {}}
```

Comment:
this will be passed up to the parent component as onSubmit (or onFinish however you want to call it)